NitroxServer/GameLogic/Bases/BaseManager.cs

```diff
@@ -33,29 +35,83 @@ public List<BasePiece> GetPartiallyConstructedPieces()
                 return new List<BasePiece>(partiallyConstructedPiecesById.Values);
             }
         }
-
         public void AddBasePiece(BasePiece basePiece)
         {
-            lock (partiallyConstructedPiecesById)
+            //anything that is not a window, hatch, or reinforcement get run normally
+            if (basePiece.TechType.Name != "BaseReinforcement" | basePiece.TechType.Name != "BaseHatch" | basePiece.TechType.Name != "BaseWindow")
+            {
+                //just in case some how a base piece is spawned at max completion
+                //the base piece will get added to at least one directory
+                if(basePiece.ConstructionAmount < 1.0f)
+                {
+                    lock (partiallyConstructedPiecesById)
+                    {
+                        partiallyConstructedPiecesById.Add(basePiece.Id, basePiece);
+                    }
+                }
+                else
+                {
+                    lock (partiallyConstructedPiecesById)
+                    {
+                        completedBasePieceHistory.Add(basePiece);
+                    }
+                }
+            }
+            //when adding base peices this tries to eliminate partially constructed parts that were once fully constructed
```

Comment:
how do you know that they were once fully constructed?

Reply:
I don't, but there is an issue in which base pieces that I personally constructed to 100% get reloaded at 89% or 95% or something else. Rather than having to deal with that I believe it would be more beneficial to just any base piece above a certain threshold, ON RELOAD. I personally do not construct a base piece and leave it at 50% or 80% and reload the world, I always finish my build beforehand and I think a lot of people can agree with me.